src/api/libcellml/componententity.h

```diff
@@ -47,11 +47,41 @@ class LIBCELLML_EXPORT ComponentEntity: public ImportedEntity
     ComponentEntity& operator=(ComponentEntity rhs); /**< Assignment operator */
 
     /**
-     * @brief Add a child Units to this component entity.
-     * Add a copy of the given Units u as a child units of this component entity.
-     * @param u The Units to add.
+     * @brief Add a child units to this component entity.
+     * Add a copy of the given units as a child units of this component entity.
+     * @param units The units to add.
      */
-    void addUnits(const UnitsPtr &u);
+    void addUnits(const UnitsPtr &units);
+
+    /**
+     * @brief Remove the units with the given name.
+     * Remove the first units found that matches the name @p name.  If the
+     * name is not found throw @c std::out_of_range.
+     * @param name The name of the units to remove.
+     */
+    void removeUnits(const std::string &name);
+
+    /**
+     * @brief Remove the units with the given pointer.
+     * Remove the first units found with the pointer @p units.  If the
+     * units are not found throw @c std::out_of_range.
+     * @overload
+     * @param units The pointer to the units to remove.
+     */
+    void removeUnits(const UnitsPtr &units);
+
+    /**
+     * @brief Remove all units stored in this component entity.
+     * Clears all units that have been added to this component entity.
+     */
+    void removeAllUnits();
+
+    /**
+     * @brief Get the number of units in the component entity.
+     * Returns the number of units the component entity directly contains.
```

Comment:
I was originally thinking about the difference between units defined at the model level as opposed to units defined in a component, but there is indeed also the case of imported units.